### ISSUE TITLE
CLI fails to run with 1.12.0

### ISSUE THREAD

**Issue by Brett-Best** [apollographql/apollo-ios#3384]
### Summary

When I attempt to run the CLI it crashes with an error.
This only occurs with 1.12.0, 1.11.0 and earlier are fine.

In CI it appears that the command just hangs with a gatekeeper dialog shown in the VM.

### Version

1.12.0

### Steps to reproduce the behavior

Download the Apollo CLI from the releases page, then try to run it in terminal from downloads:
E.g. `./apollo-ios-cli 3`

### Logs

```shell
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Incident Identifier: F6EE0328-47F4-4D48-A992-A3ABE3B1866B
CrashReporter Key:   5FC50073-92B9-0A10-A9B2-F207F8D44AF2
Hardware Model:      Mac14,10
Process:             apollo-ios-cli 3 [92059]
Path:                /Users/USER/Downloads/apollo-ios-cli 3
Identifier:          apollo-ios-cli 3
Version:             ???
Code Type:           ARM-64 (Native)
Role:                Unspecified
Parent Process:      fish [2127]
Coalition:           com.apple.Terminal [2658]
Responsible Process: Terminal [2113]

Date/Time:           2024-05-22 15:50:32.4319 +1000
Launch Time:         2024-05-22 15:50:32.3360 +1000
OS Version:          macOS 14.5 (23F79)
Release Type:        User
Report Version:      104

Exception Type:  EXC_CRASH (SIGKILL (Code Signature Invalid))
Exception Codes: 0x0000000000000000, 0x0000000000000000
Termination Reason: CODESIGNING 1 Taskgated Invalid Signature

Triggered by Thread:  0

Thread 0 Crashed:
0                                 	       0x1012c4b70 _dyld_start + 0
```

```shell
brettbest@Bretts-MacBook-Pro ~/Downloads> log stream --predicate "sender == 'AppleMobileFileIntegrity' or sender == 'AppleSystemPolicy' or process == 'amfid' or process == 'taskgated-helper' or process == 'syspolicyd'"
Filtering the log data using "sender == "AppleMobileFileIntegrity" OR sender == "AppleSystemPolicy" OR process == "amfid" OR process == "taskgated-helper" OR process == "syspolicyd""
Timestamp                       Thread     Type        Activity             PID    TTL  
2024-05-22 16:43:21.180918+1000 0x2e3815   Default     0x0                  493    0    syspolicyd: (Security) [com.apple.securityd:security_exception] MacOS error: -67062
2024-05-22 16:43:21.181677+1000 0x2e3815   Default     0x0                  493    0    syspolicyd: [com.apple.syspolicy.exec:default] GK evaluateScanResult: 1, PST: (vuid: BD786381-8AE3-4A3C-91B3-AF1402AAFC96), (objid: 248757891), (team: (null)), (id: (null)), (bundle_id: (null)), 1, 0, 1, 0, 7, 7, 0
2024-05-22 16:43:21.183016+1000 0x2e3815   Default     0x0                  493    0    syspolicyd: [com.apple.syspolicy:default] Found console users: <private>
2024-05-22 16:43:21.183130+1000 0x2e3815   Default     0x0                  493    0    syspolicyd: (Security) [com.apple.securityd:security_exception] MacOS error: -67062
2024-05-22 16:43:21.183416+1000 0x2e3815   Error       0x0                  493    0    syspolicyd: [com.apple.syspolicy:default] Unable to get certificates array: (null)
2024-05-22 16:43:21.184138+1000 0x2e3815   Default     0x0                  493    0    syspolicyd: 
2024-05-22 16:43:22.655365+1000 0x2e3a4e   Default     0x69387f             493    0    syspolicyd: [com.apple.syspolicy.exec:default] Code evaluation completed: 1
2024-05-22 16:43:22.655472+1000 0x2e3815   Default     0x0                  493    0    syspolicyd: [com.apple.syspolicy.exec:default] Adding Gatekeeper denial breadcrumb (direct): PST: (vuid: BD786381-8AE3-4A3C-91B3-AF1402AAFC96), (objid: 248757891), (team: (null)), (id: (null)), (bundle_id: (null))
2024-05-22 16:43:22.655578+1000 0x2e3815   Activity    0x694472             493    0    syspolicyd: (CoreServicesInternal) CFURLCreateBookmarkData
2024-05-22 16:43:22.659656+1000 0x2e3a4e   Error       0x0                  493    0    syspolicyd: [com.apple.syspolicy.exec:default] Terminating process due to Gatekeeper rejection: 92900, <private>
2024-05-22 16:43:22.660116+1000 0x2e3b76   Default     0x0                  0      0    kernel: (AppleSystemPolicy) ASP: Sleep interrupted: ref 1302, signal 0x100
2024-05-22 16:43:22.660123+1000 0x2e3b76   Default     0x0                  0      0    kernel: (AppleSystemPolicy) ASP: Security policy would not allow process: 92900, /Users/brettbest/Downloads/apollo-ios-cli 4
```

**Comment by calvincestari** [comment]
Thanks for reporting this @Brett-Best. I can reproduce the behaviour but I'm not sure what's going on yet. I'll get back to you once I know what's going on.

**Comment by calvincestari** [comment]
@Brett-Best this is now fixed with the `1.12.1` release. Apologies for any inconvenience.

**Comment by Brett-Best** [comment]
@calvincestari new issue raised from this https://github.com/apollographql/apollo-ios/issues/3386, appreciate the quick fix
